hs_core/serialization.py

```diff
@@ -681,22 +673,59 @@ def write_metadata_to_resource(self, resource):
         :param resource: HydroShare resource instance
         """
         for c in self.get_creators():
-            if self.owner_is_hs_user and c.order == 1:
-                # Skip the owner if the owner is a HydroShare user
-                # (i.e. they were already added when the resource was
-                # created)
-                continue
             if isinstance(c, GenericResourceMeta.ResourceCreator):
-                kwargs = {'order': c.order, 'name': c.name,
-                          'organization': c.organization,
-                          'email': c.email, 'address': c.address,
-                          'phone': c.phone, 'homepage': c.homepage,
-                          'researcherID': c.researcherID,
-                          'researchGageID': c.researchGateID}
-                try:
-                    resource.metadata.create_element('creator', **kwargs)
-                except IntegrityError:
-                    pass
+                if self.owner_is_hs_user and c.order == 1:
+                    # Use metadata from bag for owner if the owner is a HydroShare user
+                    # (because the metadata were inheritted from the user profile when we
+                    # called create_resource above)
+
+                    # Find the owner in the creators metadata
+                    owner_metadata = None
+                    for rc in resource.metadata.creators.all():
```

Comment:
@hyi Ideally I would query Django for the creator whose order == 1, but I couldn't quickly figure out how to do this with a GenericRelation.  Do you have any idea?